lib/instrumentation/core/http.js

```diff
@@ -413,37 +410,34 @@ function urlToOptions(_url) {
 }
 
 function wrapRequest(agent, request) {
-  // URL as a global was not added until Node v10
-  const URLClass = global.URL ? global.URL : url.URL
+  const URLClass = global.URL
+
   return function wrappedRequest(input, options, cb) {
     let reqArgs = arguments
-    if (SHOULD_FORMAT_ARGS) {
-      // Support new function signatures in Node v10+. If the first argument is a URL,
-      // merge it into the options object. This code is copied from Node internals.
-      if (typeof input === 'string') {
-        const urlStr = input
-        input = urlToOptions(new URLClass(urlStr))
-      } else if (input.constructor && input.constructor.name === 'URL') {
-        input = urlToOptions(input)
-      } else {
-        cb = options
-        options = input
-        input = null
-      }
 
-      if (typeof options === 'function') {
-        cb = options
-        options = input || {}
-      } else {
-        options = Object.assign(input || {}, options)
-      }
-
-      reqArgs = [options, cb]
+    // If the first argument is a URL, merge it into the options object.
+    // This code is copied from Node internals.
+    if (typeof input === 'string') {
+      const urlStr = input
+      input = urlToOptions(new URLClass(urlStr))
```

Comment:
Seems like this could be simplified to just `new URL(urlStr)` here and the URLClass definition removed.